src/main/kotlin/com/openlattice/transporter/services/TransporterService.kt

```diff
@@ -0,0 +1,125 @@
+import com.google.common.util.concurrent.ListeningExecutorService
+import com.hazelcast.core.HazelcastInstance
+import com.openlattice.assembler.AssemblerConnectionManager
+import com.openlattice.data.storage.buildPreparableFiltersSql
+import com.openlattice.datastore.services.EdmManager
+import com.openlattice.datastore.services.EntitySetManager
+import com.openlattice.edm.events.EntityTypeCreatedEvent
+import com.openlattice.edm.events.EntityTypeDeletedEvent
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.sql.Connection
+import java.util.*
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.TimeUnit
+import kotlin.system.exitProcess
+
+@Service
+final class TransporterService(
+        private val executor: ListeningExecutorService,
+        private val eventBus: EventBus,
+        private val configuration: TransporterConfiguration,
+        private val hazelcastInstance: HazelcastInstance,
+        private val entitySetService: EntitySetManager,
+        private val dataModelService: EdmManager
+)
+{
+    companion object {
+        val logger = LoggerFactory.getLogger(TransporterService::class.java)
+    }
+
+    private val transporterDataSource = AssemblerConnectionManager.createDataSource( "transporter", configuration.server, configuration.ssl )
+    private val entityTypes: MutableMap<UUID, TransporterEntityTypeState> = ConcurrentHashMap()
+
+
+    init {
+        dataModelService.entityTypes.map { et -> et.id to TransporterEntityTypeState(et, dataModelService) }.toMap(entityTypes)
+        logger.info("Creating {} entity set tables", entityTypes.size)
+        transporterDataSource.connection.use { c ->
+            entityTypes.values.forEach {
+                it.createTables(c)
+            }
+        }
+        // TODO: ensure FDW is in place
+        logger.info("Entity set tables created")
+
+        if (configuration.once) {
+            pollOnce()
+            exitProcess(0)
+        } else {
+            eventBus.register(this)
+            executor.submit {
+                pollInfinitely()
+            }
+        }
+    }
+
+    private fun pollOnce() {
+        transporterDataSource.connection.use {
+            entitySetService.getEntitySets().forEach {es ->
+//            entitySetService.getEntitySet(UUID.fromString("3a1c9bb1-5e72-4735-92b7-715cc9555eb2"))?.let { es ->
+                entityTypes[es.entityTypeId]?.updateEntitySet(it, es)
+            }
+        }
+    }
+
+    private fun pollInfinitely() {
+        Thread.currentThread().name = "transporter-change-poller"
+        while (true) {
+            try {
+                pollOnce()
+                Thread.sleep(TimeUnit.SECONDS.toMillis(1))
+            } catch (t: Throwable) {
+                logger.error("Error in poller, terminating", t)
+                exitProcess(1)
+            }
+        }
+    }
+
+    @Subscribe
+    fun handleEntityTypeCreated(e: EntityTypeCreatedEvent) {
```

Comment:
Is the plan to handle all the events, that affect materialized views in transporter? It was just my understanding before, that we keep them in Assembler and use transporter as a service to handle the call (in AssemblerConnectionManager or Assembler)
I mean, in my opinion, both are fine, we just have all that logic, hazelcast processors... already there

Reply:
I think it makes sense to put them in the same place, but I don't really care where that is. Transporter doesn't necessarily need to be it's own service, but it's just way easier for me to debug that way right now. We could turn it into a library.